omen2/object.py

```diff
@@ -446,16 +451,18 @@ def _save(self, keys: Iterable[str]):
     def __enter__(self):
         """Lock for write, and trigger thread-isolation."""
         if self.__meta and self.__meta.table:
-            if not self.__meta.lock.acquire(timeout=VERY_LARGE_LOCK_TIMEOUT):
-                log.critical("deadlock prevented", stack_info=True)
-                raise OmenLockingError
-            if self.__meta.locked:
-                # nested with blocks could work, but they are an anti-pattern
-                self.__meta.lock.release()
-                raise OmenLockingError("nested with blocks not supported")
-            self.__meta.locked = True
-            self.__meta.changes = {}
-            self.__meta.lock_id = threading.get_ident()
+            with self._table.lock:
```

Comment:
this will hold the table lock for the entire time the object is locked - obviating the need for it.   the table lock should only be held until the object lock is acquired (`self._table.locked_objs.add(self)`)